.github/workflows/sc-client-server-deb10.yml

```diff
@@ -45,19 +79,58 @@ jobs:
     runs-on: ubuntu-20.04
     steps:
     - uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
     - name: Update submodules
       run: git submodule update --init
+    
+    - name: Check what files were updated
+      id: check_changes
+      run: |
+        echo 'changed_files=$(git diff --name-only origin/HEAD | xargs)' >> $GITHUB_OUTPUT
+
+    - name: Check what Docker images have to be rebuild
+      run: |
+        changed_files=$(git diff --name-only origin/HEAD | xargs)
+        for file in "$DOCKER_BASE" "$DOCKER_REDIS" "sai.env"; do
+          if [[ "$changed_files" == *"$file"* ]]; then
+            echo "REDIS_RPC=1"
+          fi
+        done
+        for file in "$DOCKER_BASE" "$DOCKER_THRIFT" "sai.env"; do
+          if [[ "${{ steps.check_changes.outputs.changed_files }}" == *"$file"* ]]; then
+            echo "THRIFT_RPC=1" >> $GITHUB_ENV
+          fi
+        done
+
     - name: Build client Docker image
       run: ./build.sh -i client -o deb10 --nosnappi
-    - name: Build client Docker image with SAI thrift
-      run: ./build.sh -i client -s thrift -o deb10 --nosnappi
+      if: ${{ env.REDIS_RPC == '1' }}
+    - name: Build thrift client Docker image
+      run: ./build.sh -i client -o deb11 -s thrift --nosnappi
+      if: ${{ env.THRIFT_RPC == '1' }}
+    
+    - name: Pull SAI-C client
+      run: ./run.sh -i client -o deb10
+      if: ${{ env.REDIS_RPC == '0' }}
+    - name: Pull SAI-C thrift client
+      run: ./run.sh -i client -s thrift -o deb10
+      if: ${{ env.THRIFT_RPC == '0' }}
+    
     - name: Save client Docker image
       run: docker save sc-client > sc-client.tar
     - name: Upload client Docker image
       uses: actions/upload-artifact@v3
       with:
         name: Client Image
         path: sc-client.tar
+    - name: Save SAI Thrift client Docker image
```

Comment:
From what I recall, there is no saivs Thrift Server Docker file at the moment. So, this will not work.

Reply:
Removed thrift part for client/server build